.github/workflows/test.yaml

```diff
@@ -38,40 +43,84 @@ jobs:
       - name: Lint
         run: pnpm lint
 
+      - name: Validate Tests
+        run: pnpm test:validate
+
+      - name: Run Tests - ${{ matrix.env }}
+        run: pnpm ci:test --${{ matrix.env }}
+
+      - name: Upload Coverage - ${{ matrix.env }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.env }}
+          path: coverage/environments/${{ matrix.env }}
+
+  test:
```

Comment:
```suggestion
  report-coverage:
```

"Test / test (pull_request)" looks confusing in the CI checks.

Reply:
<img width="853" alt="Screenshot 2023-11-14 at 12 53 45" src="https://github.com/FuelLabs/fuels-ts/assets/9402987/dea8f96d-bc99-4077-a785-89b3fc648176">

Because of our current branch protection rules,`test` is the required last stage for us. I agree this looks wrong however I would like to handle this outside of the PR due to the changes required.